### PR TITLE
Test case for4250

### PR DESCRIPTION
**Pull request by ChamaraPhilipsuom**
IDENTITY-4250 testcase

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java [hunk 1]
```diff
@@ -52,10 +56,13 @@
 
 	private final static String SERVICE_PROVIDER_NAME = "PlaygroundServiceProver";
 	private final static String SERVICE_PROVIDER_DESC = "Playground Service Prover";
+	private static final String emailClaimURI = "http://wso2.org/claims/emailaddress";
 	private final static int TOMCAT_PORT = 8490;
 
 	protected ApplicationManagementServiceClient appMgtclient;
 	protected OauthAdminClient adminClient;
+    protected RemoteUserStoreManagerServiceClient remoteUSMServiceClient;
```
Comment (cdwijayarathna) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java]: Formatting issue

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java [hunk 2]
```diff
@@ -68,6 +75,7 @@ protected void init(TestUserMode userMode) throws Exception {
 		super.init(userMode);
 		appMgtclient = new ApplicationManagementServiceClient(sessionCookie, backendURL, null);
 		adminClient = new OauthAdminClient(backendURL, sessionCookie);
+        remoteUSMServiceClient = new RemoteUserStoreManagerServiceClient(backendURL, sessionCookie);
```
Comment (cdwijayarathna) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java]: Formatting issue

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAuthCodeGrantOpenIdTestCase.java [hunk 3]
```diff
@@ -55,210 +62,236 @@
...
+
+        consumerKey = appDto.getOauthConsumerKey();
+        Assert.assertNotNull(consumerKey, "Application creation failed.");
+        consumerSecret = appDto.getOauthConsumerSecret();
+    }
+
+    @Test(groups = "wso2.is", description = "Send authorize user request", dependsOnMethods = "testRegisterApplication")
+    public void testSendAuthorozedPost() throws Exception {
+        List<NameValuePair> urlParameters = new ArrayList<NameValuePair>();
+        urlParameters.add(new BasicNameValuePair("grantType", OAuth2Constant.OAUTH2_GRANT_TYPE_CODE));
```
Comment (cdwijayarathna) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAuthCodeGrantOpenIdTestCase.java]: Use constants for these attributes

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAuthCodeGrantOpenIdTestCase.java [hunk 4]
```diff
@@ -338,4 +371,27 @@ private void resetISConfiguration() throws Exception {
         serverConfigurationManager.restoreToLastConfiguration();
     }
 
+    public HttpResponse sendLoginPost(HttpClient client, String sessionDataKey) throws IOException {
```
Comment (cdwijayarathna) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAuthCodeGrantOpenIdTestCase.java]: Make this private if this is not used outside this class

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAuthCodeGrantOpenIdTestCase.java [hunk 5]
```diff
@@ -55,210 +62,236 @@
...
+        String valid = keyValues.get(0).getValue();
+        Assert.assertEquals(valid, "true", "Token Validation failed");
+        EntityUtils.consume(response.getEntity());
+    }
+
+    @Test(groups = "wso2.is", description = "Validate the user claim values", dependsOnMethods = "testGetAccessToken")
+    public void testClaims() throws Exception {
+        HttpGet request = new HttpGet(OAuth2Constant.USER_INFO_ENDPOINT);
+
+        request.setHeader("User-Agent", OAuth2Constant.USER_AGENT);
```
Comment (cdwijayarathna) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAuthCodeGrantOpenIdTestCase.java]: Use constants

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAuthCodeGrantOpenIdTestCase.java (hunk 4 above)
Comment (ChamaraPhilipsuom) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAuthCodeGrantOpenIdTestCase.java]: It is a override method.

**Pull request merged**